game/Utils.ts

```diff
@@ -84,6 +112,32 @@ export class Utils {
         return tree.root;
     }
 
+    /* 
+     * Constructs a proof that a given leaf (tileHash) is in the merkle root.
+     * Uses IncrementalQuinTree's genMerklePath(_index)
+     */
+    static generateMerkleProof(
+        tileHash: string,
+        mTree: IncrementalQuinTree
+    ) {
+        const h = BigNumber.from(tileHash);
+        const numLeaves = mTree.leavesPerNode ** mTree.depth;
+
+        let leafIndex = 0;
+        for (let i = 0; i < numLeaves; i++) {
+            if (h.eq(mTree.getNode(i))) {
+                leafIndex = i;
+            }
+        }
```

Comment:
should throw error if h isn't a leaf

Reply:
great responses sir. one last nit here: leafIndex should be undefined, then check if undefined. makes things crystal clear vs leafIndex -1 then check if less than 0